typegen/src/typegen/ir/module_ir.rs

```diff
@@ -9,23 +9,24 @@ use scale_info::form::PortableForm;
 
 /// Represents a Rust `mod`, containing generated types and child `mod`s.
 #[derive(Debug, Clone)]
-pub struct ModuleIR {
+pub struct ModuleIR<'a> {
     /// Name of this module.
     pub name: Ident,
     /// Root module identifier.
     pub root_mod: Ident,
     /// Submodules of this module.
-    pub children: BTreeMap<Ident, ModuleIR>,
+    pub children: BTreeMap<Ident, ModuleIR<'a>>,
     /// Types in this module.
-    pub types: BTreeMap<scale_info::Path<PortableForm>, TypeIR>,
+    pub types:
+        BTreeMap<scale_info::Path<PortableForm>, (&'a scale_info::Type<PortableForm>, TypeIR)>,
```

Comment:
I wonder whether `scale_info::Path<PortableForm>` can also be a reference now? Not if you need to alter the path here for de-duping but possible if you've already de-duped by then perhaps?